File: String/1578. Minimum Time to Make Rope Colorful.cpp
class Solution {
public:
    int minCost(string c, vector<int>& nt) 
    {
        int n = c.size();
        int ans = 0, prev = 0;
        for(int i = 1; i < n; i++)
        {
            if(c[prev] != c[i]) prev = i;
            else 
            {
                if(nt[prev] < nt[i])
                {
                    ans += nt[prev];
                    prev = i;
                }
                else ans += nt[i];
            }
        }
        return ans;
    }
};

File: Array/1578. Minimum Time to Make Rope Colorful.py
class Solution:
    def minCost(self, colors: str, neededTime: List[int]) -> int:
        ans, i, n = 0, 0, len(colors)
        while i < n:
            j = i + 1
            time = [neededTime[i]]
            while j < n and colors[j] == colors[i]:
                time.append(neededTime[j])
                j += 1
            i = j
            ans += sum(time) - max(time)
        return ans